DX: misleading error messages

using a invalid configuration like (note `names:` instead of `value:`)

```
  layers:
    - name: SuperGlobal
      collectors:
        - type: superglobal
          names:
            - _POST
            - _GET
            - _REQUEST
```

leads to a error like

```
$ composer deptrac
> vendor/bin/deptrac analyze --config-file=app/bridge/deptrac.yaml --report-uncovered --fail-on-uncovered
   1/272 [>---------------------------]   0%


 [ERROR] Analysis finished with an Exception.

         Invalid collector definition.

         SuperglobalCollector needs the names configuration.
```

the above config was valid in deptrac 1.x and I think the error message made sense at that time.
since deptrac 2.x requires a `value:` config attribute, I think the error message is no longer fitting:

> SuperglobalCollector needs the names configuration.

---

I had a similar experience with 
```
    - name: Controller
      collectors:
        - type: classLike
          regex: .*Controller.*(?<!ActionController)
```

which worked in deptrac 1.x but leads to this misleading error on deptrac 2.x:
```
$ composer deptrac
> vendor/bin/deptrac analyze --config-file=app/bridge/deptrac.yaml --report-uncovered --fail-on-uncovered



 [ERROR] Analysis finished with an Exception.

         Invalid collector definition.

         Collector "classLike" needs the regex configuration.
```

--- a/src/Contract/Layer/InvalidLayerDefinitionException.php
+++ b/src/Contract/Layer/InvalidLayerDefinitionException.php
@@ -36,8 +36,8 @@ final class InvalidLayerDefinitionException extends RuntimeException implements 
         return new self('Layer configuration is empty. You need to define at least 1 layer.');
     }
 
-    public static function circularTokenReference(string $tokenName): self
+    public static function circularTokenReference(string $collectorName, string $tokenName): self
     {
-        return new self(sprintf('Circular dependency between layers detected. Token "%s" could not be resolved.', $tokenName));
+        return new self(sprintf('%s: Circular dependency between layers detected. Token "%s" could not be resolved.', $collectorName, $tokenName));
     }
 }

--- a/src/DefaultBehavior/Layer/AttributeCollector.php
+++ b/src/DefaultBehavior/Layer/AttributeCollector.php
@@ -49,8 +49,11 @@ final class AttributeCollector implements CollectorInterface
      */
     private function getSearchedSubstring(array $config): string
     {
-        if (!isset($config['value']) || !is_string($config['value'])) {
-            throw InvalidCollectorDefinitionException::invalidCollectorConfiguration('AttributeCollector needs the attribute name as a string.');
+        if (!isset($config['value'])) {
+            throw InvalidCollectorDefinitionException::invalidCollectorConfiguration('AttributeCollector: Missing configuration.');
+        }
+        if (!is_string($config['value'])) {
+            throw InvalidCollectorDefinitionException::invalidCollectorConfiguration('AttributeCollector: Configuration is not a string.');
         }
 
         return $config['value'];

--- a/src/DefaultBehavior/Layer/BoolCollector.php
+++ b/src/DefaultBehavior/Layer/BoolCollector.php
@@ -58,7 +58,7 @@ final class BoolCollector implements CollectorInterface
         }
 
         if (!$configuration['must'] && !$configuration['must_not']) {
-            throw InvalidCollectorDefinitionException::invalidCollectorConfiguration('"bool" collector must have a "must" or a "must_not" attribute.');
+            throw InvalidCollectorDefinitionException::invalidCollectorConfiguration('BoolCollector: Missing a "must" or a "must_not" attribute.');
         }
 
         return $configuration;

--- a/src/DefaultBehavior/Layer/ClassNameRegexCollector.php
+++ b/src/DefaultBehavior/Layer/ClassNameRegexCollector.php
@@ -6,7 +6,6 @@ namespace Deptrac\Deptrac\DefaultBehavior\Layer;
 
 use Deptrac\Deptrac\Contract\Ast\AstMap\ClassLikeReference;
 use Deptrac\Deptrac\Contract\Ast\AstMap\TokenReferenceInterface;
-use Deptrac\Deptrac\Contract\Layer\InvalidCollectorDefinitionException;
 use Deptrac\Deptrac\DefaultBehavior\Layer\Helpers\RegexCollector;
 
 final class ClassNameRegexCollector extends RegexCollector
@@ -20,12 +19,8 @@ final class ClassNameRegexCollector extends RegexCollector
         return $reference->getToken()->match($this->getValidatedPattern($config));
     }
 
-    protected function getPattern(array $config): string
+    protected function getPattern(string $config): string
     {
-        if (!isset($config['value']) || !is_string($config['value'])) {
-            throw InvalidCollectorDefinitionException::invalidCollectorConfiguration('ClassNameRegexCollector needs the regex configuration.');
-        }
-
-        return $config['value'];
+        return $config;
     }
 }

--- a/src/DefaultBehavior/Layer/ComposerCollector.php
+++ b/src/DefaultBehavior/Layer/ComposerCollector.php
@@ -21,28 +21,28 @@ final class ComposerCollector implements CollectorInterface
     public function satisfy(array $config, TokenReferenceInterface $reference): bool
     {
         if (!isset($config['composerPath']) || !is_string($config['composerPath'])) {
-            throw InvalidCollectorDefinitionException::invalidCollectorConfiguration('ComposerCollector needs the path to the composer.json file as string.');
+            throw InvalidCollectorDefinitionException::invalidCollectorConfiguration('ComposerCollector: path to the "composer.json" file is not a string.');
         }
 
         if (!isset($config['composerLockPath']) || !is_string($config['composerLockPath'])) {
-            throw InvalidCollectorDefinitionException::invalidCollectorConfiguration('ComposerCollector needs the path to the composer.lock file as string.');
+            throw InvalidCollectorDefinitionException::invalidCollectorConfiguration('ComposerCollector: path to the "composer.lock" file is not a string.');
         }
 
         if (!isset($config['packages']) || !is_array($config['packages'])) {
-            throw InvalidCollectorDefinitionException::invalidCollectorConfiguration('ComposerCollector needs the list of packages as string.');
+            throw InvalidCollectorDefinitionException::invalidCollectorConfiguration('ComposerCollector: "packages" is not an array.');
         }
 
         try {
             $this->parser[$config['composerLockPath']] ??= new ComposerFilesParser($config['composerLockPath']);
             $parser = $this->parser[$config['composerLockPath']];
         } catch (RuntimeException $exception) {
-            throw new CouldNotParseFileException('Could not parse composer files.', 0, $exception);
+            throw new CouldNotParseFileException('ComposerCollector: Could not parse composer files.', 0, $exception);
         }
 
         try {
             $namespaces = $parser->autoloadableNamespacesForRequirements($config['packages'], true);
         } catch (RuntimeException $e) {
-            throw InvalidCollectorDefinitionException::invalidCollectorConfiguration(sprintf('ComposerCollector has a non-existent package defined. %s', $e->getMessage()));
+            throw InvalidCollectorDefinitionException::invalidCollectorConfiguration(sprintf('ComposerCollector: Non-existent package defined. %s', $e->getMessage()));
         }
 
         $token = $reference->getToken()->toString();

--- a/src/DefaultBehavior/Layer/DirectoryCollector.php
+++ b/src/DefaultBehavior/Layer/DirectoryCollector.php
@@ -5,7 +5,6 @@ declare(strict_types=1);
 namespace Deptrac\Deptrac\DefaultBehavior\Layer;
 
 use Deptrac\Deptrac\Contract\Ast\AstMap\TokenReferenceInterface;
-use Deptrac\Deptrac\Contract\Layer\InvalidCollectorDefinitionException;
 use Deptrac\Deptrac\DefaultBehavior\Layer\Helpers\RegexCollector;
 use Symfony\Component\Filesystem\Path;
 
@@ -25,12 +24,8 @@ final class DirectoryCollector extends RegexCollector
         return 1 === preg_match($validatedPattern, $normalizedPath);
     }
 
-    protected function getPattern(array $config): string
+    protected function getPattern(string $config): string
     {
-        if (!isset($config['value']) || !is_string($config['value'])) {
-            throw InvalidCollectorDefinitionException::invalidCollectorConfiguration('DirectoryCollector needs the regex configuration.');
-        }
-
-        return '#'.$config['value'].'#i';
+        return '#'.$config.'#i';
     }
 }

--- a/src/DefaultBehavior/Layer/ExtendsCollector.php
+++ b/src/DefaultBehavior/Layer/ExtendsCollector.php
@@ -50,8 +50,11 @@ final class ExtendsCollector implements CollectorInterface
      */
     private function getInterfaceName(array $config): ClassLikeToken
     {
-        if (!isset($config['value']) || !is_string($config['value'])) {
-            throw InvalidCollectorDefinitionException::invalidCollectorConfiguration('ExtendsCollector needs the interface or class name as a string.');
+        if (!isset($config['value'])) {
+            throw InvalidCollectorDefinitionException::invalidCollectorConfiguration('ExtendsCollector: Missing configuration.');
+        }
+        if (!is_string($config['value'])) {
+            throw InvalidCollectorDefinitionException::invalidCollectorConfiguration('ExtendsCollector: Configuration is not a string.');
         }
 
         return ClassLikeToken::fromFQCN($config['value']);

--- a/src/DefaultBehavior/Layer/FunctionNameCollector.php
+++ b/src/DefaultBehavior/Layer/FunctionNameCollector.php
@@ -31,8 +31,11 @@ final class FunctionNameCollector implements CollectorInterface
      */
     private function getPattern(array $config): string
     {
-        if (!isset($config['value']) || !is_string($config['value'])) {
-            throw InvalidCollectorDefinitionException::invalidCollectorConfiguration('FunctionNameCollector needs the regex configuration.');
+        if (!isset($config['value'])) {
+            throw InvalidCollectorDefinitionException::invalidCollectorConfiguration('FunctionNameCollector: Missing configuration.');
+        }
+        if (!is_string($config['value'])) {
+            throw InvalidCollectorDefinitionException::invalidCollectorConfiguration('FunctionNameCollector: Configuration is not a string.');
         }
 
         return '/'.$config['value'].'/i';

--- a/src/DefaultBehavior/Layer/GlobCollector.php
+++ b/src/DefaultBehavior/Layer/GlobCollector.php
@@ -5,7 +5,6 @@ declare(strict_types=1);
 namespace Deptrac\Deptrac\DefaultBehavior\Layer;
 
 use Deptrac\Deptrac\Contract\Ast\AstMap\TokenReferenceInterface;
-use Deptrac\Deptrac\Contract\Layer\InvalidCollectorDefinitionException;
 use Deptrac\Deptrac\DefaultBehavior\Layer\Helpers\RegexCollector;
 use Symfony\Component\Filesystem\Path;
 use Symfony\Component\Finder\Glob;
@@ -35,12 +34,8 @@ final class GlobCollector extends RegexCollector
         return 1 === preg_match($validatedPattern, $relativeFilePath);
     }
 
-    protected function getPattern(array $config): string
+    protected function getPattern(string $config): string
     {
-        if (!isset($config['value']) || !is_string($config['value'])) {
-            throw InvalidCollectorDefinitionException::invalidCollectorConfiguration('GlobCollector needs the glob pattern configuration.');
-        }
-
-        return Glob::toRegex($config['value']);
+        return Glob::toRegex($config);
     }
 }

--- a/src/DefaultBehavior/Layer/Helpers/AbstractTypeCollector.php
+++ b/src/DefaultBehavior/Layer/Helpers/AbstractTypeCollector.php
@@ -7,9 +7,6 @@ namespace Deptrac\Deptrac\DefaultBehavior\Layer\Helpers;
 use Deptrac\Deptrac\Contract\Ast\AstMap\ClassLikeReference;
 use Deptrac\Deptrac\Contract\Ast\AstMap\ClassLikeType;
 use Deptrac\Deptrac\Contract\Ast\AstMap\TokenReferenceInterface;
-use Deptrac\Deptrac\Contract\Layer\InvalidCollectorDefinitionException;
-
-use function is_string;
 
 abstract class AbstractTypeCollector extends RegexCollector
 {
@@ -27,12 +24,8 @@ abstract class AbstractTypeCollector extends RegexCollector
         return ($isClassLike || $isSameType) && $reference->getToken()->match($this->getValidatedPattern($config));
     }
 
-    protected function getPattern(array $config): string
+    protected function getPattern(string $config): string
     {
-        if (!isset($config['value']) || !is_string($config['value'])) {
-            throw InvalidCollectorDefinitionException::invalidCollectorConfiguration(sprintf('Collector "%s" needs the regex configuration.', $this->getType()->toString()));
-        }
-
-        return '/'.$config['value'].'/i';
+        return '/'.$config.'/i';
     }
 }

--- a/src/DefaultBehavior/Layer/Helpers/RegexCollector.php
+++ b/src/DefaultBehavior/Layer/Helpers/RegexCollector.php
@@ -10,11 +10,9 @@ use Deptrac\Deptrac\Contract\Layer\InvalidCollectorDefinitionException;
 abstract class RegexCollector implements CollectorInterface
 {
     /**
-     * @param array<string, bool|string|array<string, string>> $config
-     *
      * @throws InvalidCollectorDefinitionException
      */
-    abstract protected function getPattern(array $config): string;
+    abstract protected function getPattern(string $config): string;
 
     /**
      * @param array<string, bool|string|array<string, string>> $config
@@ -23,10 +21,24 @@ abstract class RegexCollector implements CollectorInterface
      */
     protected function getValidatedPattern(array $config): string
     {
-        $pattern = $this->getPattern($config);
+        if (!isset($config['value'])) {
+            throw InvalidCollectorDefinitionException::invalidCollectorConfiguration(static::getClassName().': Missing configuration.');
+        }
+        if (!is_string($config['value'])) {
+            throw InvalidCollectorDefinitionException::invalidCollectorConfiguration(static::getClassName().': Configuration is not a string.');
+        }
+
+        $pattern = $this->getPattern($config['value']);
         if (false !== @preg_match($pattern, '')) {
             return $pattern;
         }
-        throw InvalidCollectorDefinitionException::invalidCollectorConfiguration('Invalid regex pattern '.$pattern);
+        throw InvalidCollectorDefinitionException::invalidCollectorConfiguration(static::getClassName().': Invalid regex pattern '.$pattern);
+    }
+
+    protected static function getClassName(): string
+    {
+        $lastPart = strrchr(static::class, '\\');
+
+        return false === $lastPart ? static::class : substr($lastPart, 1);
     }
 }

--- a/src/DefaultBehavior/Layer/ImplementsCollector.php
+++ b/src/DefaultBehavior/Layer/ImplementsCollector.php
@@ -50,8 +50,11 @@ final class ImplementsCollector implements CollectorInterface
      */
     private function getInterfaceName(array $config): ClassLikeToken
     {
-        if (!isset($config['value']) || !is_string($config['value'])) {
-            throw InvalidCollectorDefinitionException::invalidCollectorConfiguration('ImplementsCollector needs the interface name as a string.');
+        if (!isset($config['value'])) {
+            throw InvalidCollectorDefinitionException::invalidCollectorConfiguration('ImplementsCollector: Missing configuration.');
+        }
+        if (!is_string($config['value'])) {
+            throw InvalidCollectorDefinitionException::invalidCollectorConfiguration('ImplementsCollector: Configuration is not a string.');
         }
 
         return ClassLikeToken::fromFQCN($config['value']);

--- a/src/DefaultBehavior/Layer/InheritanceLevelCollector.php
+++ b/src/DefaultBehavior/Layer/InheritanceLevelCollector.php
@@ -31,8 +31,12 @@ final class InheritanceLevelCollector implements CollectorInterface
             throw CouldNotParseFileException::because('Could not build Ast map', $exception);
         }
         $classInherits = $this->astMap->getClassInherits($reference->getToken());
-        if (!isset($config['value']) || !is_numeric($config['value'])) {
-            throw InvalidCollectorDefinitionException::invalidCollectorConfiguration('InheritanceLevelCollector needs inheritance depth as int.');
+
+        if (!isset($config['value'])) {
+            throw InvalidCollectorDefinitionException::invalidCollectorConfiguration('InheritanceLevelCollector: Missing configuration.');
+        }
+        if (!is_numeric($config['value'])) {
+            throw InvalidCollectorDefinitionException::invalidCollectorConfiguration('InheritanceLevelCollector: Configuration is not a number.');
         }
 
         $depth = (int) $config['value'];

--- a/src/DefaultBehavior/Layer/InheritsCollector.php
+++ b/src/DefaultBehavior/Layer/InheritsCollector.php
@@ -49,8 +49,11 @@ final class InheritsCollector implements CollectorInterface
      */
     private function getClassLikeName(array $config): ClassLikeToken
     {
-        if (!isset($config['value']) || !is_string($config['value'])) {
-            throw InvalidCollectorDefinitionException::invalidCollectorConfiguration('InheritsCollector needs the interface, trait or class name as a string.');
+        if (!isset($config['value'])) {
+            throw InvalidCollectorDefinitionException::invalidCollectorConfiguration('InheritsCollector: Missing configuration.');
+        }
+        if (!is_string($config['value'])) {
+            throw InvalidCollectorDefinitionException::invalidCollectorConfiguration('InheritsCollector: Configuration is not a string.');
         }
 
         return ClassLikeToken::fromFQCN($config['value']);

--- a/src/DefaultBehavior/Layer/LayerCollector.php
+++ b/src/DefaultBehavior/Layer/LayerCollector.php
@@ -25,19 +25,23 @@ final class LayerCollector implements CollectorInterface
 
     public function satisfy(array $config, TokenReferenceInterface $reference): bool
     {
-        if (!isset($config['value']) || !is_string($config['value'])) {
-            throw InvalidCollectorDefinitionException::invalidCollectorConfiguration('LayerCollector needs the layer configuration, expected \'value\' config is missing or invalid.');
+        if (!isset($config['value'])) {
+            throw InvalidCollectorDefinitionException::invalidCollectorConfiguration('LayerCollector: Missing configuration.');
         }
+        if (!is_string($config['value'])) {
+            throw InvalidCollectorDefinitionException::invalidCollectorConfiguration('LayerCollector: Configuration is not a string.');
+        }
+
         $layer = $config['value'];
 
         if (!$this->resolver->has($layer)) {
-            throw InvalidCollectorDefinitionException::invalidCollectorConfiguration(sprintf('Unknown layer "%s" specified in collector.', $config['value']));
+            throw InvalidCollectorDefinitionException::invalidCollectorConfiguration(sprintf('LayerCollector: Unknown layer "%s" specified in collector.', $layer));
         }
         $token = $reference->getToken()->toString();
 
         if (array_key_exists($token, $this->resolved) && array_key_exists($layer, $this->resolved[$token])) {
             if (null === $this->resolved[$token][$layer]) {
-                throw InvalidLayerDefinitionException::circularTokenReference($token);
+                throw InvalidLayerDefinitionException::circularTokenReference('LayerCollector', $token);
             }
 
             return $this->resolved[$token][$layer];

--- a/src/DefaultBehavior/Layer/MethodCollector.php
+++ b/src/DefaultBehavior/Layer/MethodCollector.php
@@ -7,7 +7,6 @@ namespace Deptrac\Deptrac\DefaultBehavior\Layer;
 use Deptrac\Deptrac\Contract\Ast\AstMap\ClassLikeReference;
 use Deptrac\Deptrac\Contract\Ast\AstMap\TokenReferenceInterface;
 use Deptrac\Deptrac\Contract\Ast\ParserInterface;
-use Deptrac\Deptrac\Contract\Layer\InvalidCollectorDefinitionException;
 use Deptrac\Deptrac\DefaultBehavior\Layer\Helpers\RegexCollector;
 
 final class MethodCollector extends RegexCollector
@@ -33,12 +32,8 @@ final class MethodCollector extends RegexCollector
         return false;
     }
 
-    protected function getPattern(array $config): string
+    protected function getPattern(string $config): string
     {
-        if (!isset($config['value']) || !is_string($config['value'])) {
-            throw InvalidCollectorDefinitionException::invalidCollectorConfiguration('MethodCollector needs the name configuration.');
-        }
-
-        return '/'.$config['value'].'/i';
+        return '/'.$config.'/i';
     }
 }

--- a/src/DefaultBehavior/Layer/PhpInternalCollector.php
+++ b/src/DefaultBehavior/Layer/PhpInternalCollector.php
@@ -48,8 +48,11 @@ final class PhpInternalCollector implements CollectorInterface
      */
     private function getPattern(array $config): string
     {
-        if (!isset($config['value']) || !is_string($config['value'])) {
-            throw InvalidCollectorDefinitionException::invalidCollectorConfiguration('PhpInternalCollector needs configuration.');
+        if (!isset($config['value'])) {
+            throw InvalidCollectorDefinitionException::invalidCollectorConfiguration('PhpInternalCollector: Missing configuration.');
+        }
+        if (!is_string($config['value'])) {
+            throw InvalidCollectorDefinitionException::invalidCollectorConfiguration('PhpInternalCollector: Configuration is not a string.');
         }
 
         return '/'.$config['value'].'/i';

--- a/src/DefaultBehavior/Layer/SuperglobalCollector.php
+++ b/src/DefaultBehavior/Layer/SuperglobalCollector.php
@@ -29,8 +29,11 @@ final class SuperglobalCollector implements CollectorInterface
      */
     private function getNames(array $config): array
     {
-        if (!isset($config['value']) || !is_array($config['value'])) {
-            throw InvalidCollectorDefinitionException::invalidCollectorConfiguration('SuperglobalCollector needs the names configuration.');
+        if (!isset($config['value'])) {
+            throw InvalidCollectorDefinitionException::invalidCollectorConfiguration('SuperglobalCollector: Missing configuration.');
+        }
+        if (!is_array($config['value'])) {
+            throw InvalidCollectorDefinitionException::invalidCollectorConfiguration('SuperglobalCollector: Configuration is not an array.');
         }
 
         return array_map(static fn ($name): string => '$'.$name, $config['value']);

--- a/src/DefaultBehavior/Layer/TagValueRegexCollector.php
+++ b/src/DefaultBehavior/Layer/TagValueRegexCollector.php
@@ -6,10 +6,10 @@ namespace Deptrac\Deptrac\DefaultBehavior\Layer;
 
 use Deptrac\Deptrac\Contract\Ast\AstMap\TaggedTokenReferenceInterface;
 use Deptrac\Deptrac\Contract\Ast\AstMap\TokenReferenceInterface;
+use Deptrac\Deptrac\Contract\Layer\CollectorInterface;
 use Deptrac\Deptrac\Contract\Layer\InvalidCollectorDefinitionException;
-use Deptrac\Deptrac\DefaultBehavior\Layer\Helpers\RegexCollector;
 
-final class TagValueRegexCollector extends RegexCollector
+final class TagValueRegexCollector implements CollectorInterface
 {
     /**
      * @param array<string, bool|string|array<string, string>> $config
@@ -43,27 +43,45 @@ final class TagValueRegexCollector extends RegexCollector
      */
     protected function getTagName(array $config): string
     {
-        if (!isset($config['tag']) || !is_string($config['tag'])) {
-            throw InvalidCollectorDefinitionException::invalidCollectorConfiguration('TagValueRegexCollector needs the tag name.');
+        if (!isset($config['tag'])) {
+            throw InvalidCollectorDefinitionException::invalidCollectorConfiguration('TagValueRegexCollector: Missing "tag" configuration.');
+        }
+        if (!is_string($config['tag'])) {
+            throw InvalidCollectorDefinitionException::invalidCollectorConfiguration('TagValueRegexCollector: Configuration "tag" is not a string.');
         }
 
         if (!preg_match('/^@[-\w]+$/', $config['tag'])) {
-            throw InvalidCollectorDefinitionException::invalidCollectorConfiguration('TagValueRegexCollector needs a valid tag name.');
+            throw InvalidCollectorDefinitionException::invalidCollectorConfiguration('TagValueRegexCollector: Invalid "tag" name.');
         }
 
         return $config['tag'];
     }
 
-    protected function getPattern(array $config): string
+    /**
+     * @param array<string, bool|string|array<string, string>> $config
+     *
+     * @throws InvalidCollectorDefinitionException
+     */
+    protected function getValidatedPattern(array $config): string
     {
         if (!isset($config['value'])) {
             return '/^.?/'; // any string
         }
 
-        if (!is_string($config['value'])) {
-            throw InvalidCollectorDefinitionException::invalidCollectorConfiguration('TagValueRegexCollector regex value must be a string.');
+        $pattern = $config['value'];
+
+        if (!is_string($pattern)) {
+            throw InvalidCollectorDefinitionException::invalidCollectorConfiguration('TagValueRegexCollector: "value" configuration is not a string.');
         }
 
-        return $config['value'];
+        if ('' === $pattern) {
+            throw InvalidCollectorDefinitionException::invalidCollectorConfiguration('TagValueRegexCollector: "value" configuration is empty string.');
+        }
+
+        if (false === @preg_match($pattern, '')) {
+            throw InvalidCollectorDefinitionException::invalidCollectorConfiguration('TagValueRegexCollector: Invalid regex pattern '.$pattern);
+        }
+
+        return $pattern;
     }
 }

--- a/src/DefaultBehavior/Layer/UsesCollector.php
+++ b/src/DefaultBehavior/Layer/UsesCollector.php
@@ -50,8 +50,11 @@ final class UsesCollector implements CollectorInterface
      */
     private function getTraitName(array $config): ClassLikeToken
     {
-        if (!isset($config['value']) || !is_string($config['value'])) {
-            throw InvalidCollectorDefinitionException::invalidCollectorConfiguration('UsesCollector needs the trait name as a string.');
+        if (!isset($config['value'])) {
+            throw InvalidCollectorDefinitionException::invalidCollectorConfiguration('UsesCollector: Missing configuration.');
+        }
+        if (!is_string($config['value'])) {
+            throw InvalidCollectorDefinitionException::invalidCollectorConfiguration('UsesCollector: Configuration is not a string.');
         }
 
         return ClassLikeToken::fromFQCN($config['value']);

--- a/tests/Core/Layer/Collector/BoolCollectorTest.php
+++ b/tests/Core/Layer/Collector/BoolCollectorTest.php
@@ -226,7 +226,7 @@ final class BoolCollectorTest extends TestCase
         $reference = new ClassLikeReference(ClassLikeToken::fromFQCN('App\\Foo'));
 
         $this->expectException(InvalidCollectorDefinitionException::class);
-        $this->expectExceptionMessage('"bool" collector must have a "must" or a "must_not" attribute.');
+        $this->expectExceptionMessage('BoolCollector: Missing a "must" or a "must_not" attribute.');
 
         $this->collector->satisfy($config, $reference);
     }

--- a/tests/Core/Layer/Collector/DirectoryCollectorTest.php
+++ b/tests/Core/Layer/Collector/DirectoryCollectorTest.php
@@ -52,7 +52,7 @@ final class DirectoryCollectorTest extends TestCase
         $fileReference = $fileReferenceBuilder->build();
 
         $this->expectException(InvalidCollectorDefinitionException::class);
-        $this->expectExceptionMessage('DirectoryCollector needs the regex configuration.');
+        $this->expectExceptionMessage('DirectoryCollector: Missing configuration.');
 
         $this->collector->satisfy(
             [],

--- a/tests/Core/Layer/Collector/LayerCollectorTest.php
+++ b/tests/Core/Layer/Collector/LayerCollectorTest.php
@@ -29,7 +29,7 @@ final class LayerCollectorTest extends TestCase
     public function testConfig(): void
     {
         $this->expectException(InvalidCollectorDefinitionException::class);
-        $this->expectExceptionMessage('LayerCollector needs the layer configuration');
+        $this->expectExceptionMessage('LayerCollector: Missing configuration');
 
         $this->collector->satisfy(
             [],
@@ -47,7 +47,7 @@ final class LayerCollectorTest extends TestCase
         ;
 
         $this->expectException(InvalidCollectorDefinitionException::class);
-        $this->expectExceptionMessage('Unknown layer "test" specified in collector.');
+        $this->expectExceptionMessage('LayerCollector: Unknown layer "test" specified in collector.');
 
         $this->collector->satisfy(
             ['value' => 'test'],
@@ -72,7 +72,7 @@ final class LayerCollectorTest extends TestCase
         ;
 
         $this->expectException(InvalidLayerDefinitionException::class);
-        $this->expectExceptionMessage('Circular dependency between layers detected. Token "App\Foo" could not be resolved.');
+        $this->expectExceptionMessage('LayerCollector: Circular dependency between layers detected. Token "App\Foo" could not be resolved.');
 
         $this->collector->satisfy(
             ['value' => 'FooLayer'],

--- a/tests/Core/Layer/Collector/MethodCollectorTest.php
+++ b/tests/Core/Layer/Collector/MethodCollectorTest.php
@@ -112,7 +112,7 @@ final class MethodCollectorTest extends TestCase
         $astClassReference = new ClassLikeReference(ClassLikeToken::fromFQCN('foo'));
 
         $this->expectException(InvalidCollectorDefinitionException::class);
-        $this->expectExceptionMessage('MethodCollector needs the name configuration.');
+        $this->expectExceptionMessage('MethodCollector: Missing configuration.');
 
         $this->collector->satisfy(
             [],

